### PR TITLE
Improve code quality check processes

### PR DESCRIPTION
**Pull request by szkl**
Signed-off-by: Sonmez Kartal <sonmez@koding.com>

### CODE REVIEW

File: scripts/check_socialapi_code_quality.sh [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
+
+export GOPATH=${GOPATH:-$(pwd)/go}
+
+echo "checking deadcode"
+ls --directory $GOPATH/src/socialapi/*/**/**/ | \
+  xargs go/bin/deadcode
+
+echo "checking unused variables"
+echo $(cd $GOPATH/src; ls --directory socialapi/*/**/**/) | \
+  xargs $GOPATH/bin/varcheck
```
Comment (rjeczalik) [inline scripts/check_socialapi_code_quality.sh]: How about:

```
find $GOPATH/src/socialapi -type d -exec $GOPATH/bin/varcheck {} \;
```

Comment (szkl) [inline scripts/check_socialapi_code_quality.sh]: That's what came to my mind first too. I assumed there's a catch for third level subdirectories.

Comment (rjeczalik) [inline scripts/check_socialapi_code_quality.sh]: I think 3 was the max depth at the time of writing this. 🤔

Comment (cihangir) [inline scripts/check_socialapi_code_quality.sh]: @rjeczalik  was right, 3 was the max depth i believe. `go list` would be better tho

**Review by cihangir — APPROVED** [review]

**Review by rjeczalik — APPROVED** [review]

**Pull request merged**